File: Algorithms/src/2840. Check if Strings Can be Made Equal With Operations II/Code/solution.java
class Solution {
    public boolean checkStrings(String s1, String s2) {
        // Frequency arrays for even and odd positions
        int[] even = new int[26];
        int[] odd = new int[26];

        for (int i = 0; i < s1.length(); i++) {
            if (i % 2 == 0) {
                // Count characters at even indexes
                even[s1.charAt(i) - 'a']++;
                even[s2.charAt(i) - 'a']--;
            } else {
                // Count characters at odd indexes
                odd[s1.charAt(i) - 'a']++;
                odd[s2.charAt(i) - 'a']--;
            }
        }

        // Check if all frequencies become zero
        for (int i = 0; i < 26; i++) {
            if (even[i] != 0 || odd[i] != 0) {
                return false;
            }
        }

        return true;
    }
}

File: Algorithms/src/2840. Check if Strings Can be Made Equal With Operations II/Code/solution.py
class Solution:
    def checkStrings(self, s1: str, s2: str) -> bool:
        # Frequency arrays for even and odd positions
        even = [0] * 26
        odd = [0] * 26

        for i in range(len(s1)):
            if i % 2 == 0:
                # Count characters at even indexes
                even[ord(s1[i]) - ord('a')] += 1
                even[ord(s2[i]) - ord('a')] -= 1
            else:
                # Count characters at odd indexes
                odd[ord(s1[i]) - ord('a')] += 1
                odd[ord(s2[i]) - ord('a')] -= 1

        # Check if all frequencies become zero
        for i in range(26):
            if even[i] != 0 or odd[i] != 0:
                return False

        return True